mlos_core/mlos_core/optimizers/bayesian_optimizers.py

```diff
@@ -39,7 +40,7 @@ def surrogate_predict(self, configurations: pd.DataFrame,
 
     @abstractmethod
     def acquisition_function(self, configurations: pd.DataFrame,
-                             context: Optional[pd.DataFrame] = None) -> Callable:
+                             context: Optional[pd.DataFrame] = None) -> npt.NDArray:
```

Comment:
Hmm, maybe we should rename this to `aquire` instead of `aquisition_function` since it invokes it and doesn't just return the function.

Reply:
@bpkroth I agree, but OTOH `acquisition_function` is such a common term in Bayesian optimization that it may make sense to keep the original name. At any rate, I would not do it in this PR

Reply:
Agreed, it's just a semantics thing about what the method returns.  If it's an `@property` that returns a `Callable`, then `acquisition_function` that can be called makes sense to me.  Else, `invoke_acquisition_function` makes more sense to me.

Anyways, forked off to another issue here: #341